.github/workflows/build.yml

```diff
@@ -0,0 +1,20 @@
+name: CI
+on:
+  release:
+    types: [released, prereleased]
+  workflow_dispatch:  # allow manually running from the Actions tab
```

Comment:
@stevearc fwiw you as maintainer can trigger the flow manually for any branch from the Actions tab.

did the acript fail locally or in CI?

Reply:
Failed in CI https://github.com/stevearc/pypicloud-docker/runs/7966046266?check_suite_focus=true

Reply:
it [wants](https://github.com/stevearc/pypicloud-docker/runs/7966046266?check_suite_focus=true#step:6:2588) an older version of `psycopg-binary`, which did not have `aarch64_manylinux2014` wheels available [yet](https://pypi.org/project/psycopg-binary/3.0.16/#files) 🤔

Reply:
correction: just saw `psycopg2-binary`, with the 2. none of those have arm wheels available ;( solutions would be to add support for v3 (maybe no breaking changes that hit us), or prepare the dockerfile so the v2 install from source will succeed

Reply:
could try [adding](https://www.psycopg.org/docs/install.html#build-prerequisites) the apt package libpq-dev and see if source ibstall in the arm image starts working

Reply:
Adding `libpq-dev` fixed the issue with psycopg2. Ran into some other issues with the rust toolchain for cryptography and bcrypt, but managed to work around some of it. Current status is that it compiles for 3.5 hours and then [fails on bcrypt](https://github.com/stevearc/pypicloud-docker/runs/7999125504?check_suite_focus=true#step:6:20945)

Reply:
3.5hrs... D:

looks like adding 32bit arm (v7) into the mix might've been wishful thinking...

building wheels apparently is painfully slow for v7. probably easy fix for v7 is adding:

```
ENV PIP_EXTRA_INDEX_URL=https://www.piwheels.org/simple
```

or simply remove v7 altogether 😅

Reply:
they have wheels for v3 (v4 was released yesterday, currently failing and might be unwanted even)

https://www.piwheels.org/project/bcrypt/

Reply:
The piwheel for arm v7 might work for the phusion/baseimage build, but I would be shocked if it worked properly in alpine linux. If it's not a strict requirement for you, I'll just try it again without the v7 arm platform.

Reply:
totally fine! v7 only represents a tiny portion of users anyway [ref](https://github.com/giampaolo/psutil/pull/2103)

Reply:
> v4 was released yesterday

nonetheless maybe good to pin `bcrypt<4` to avoid potential runtime bugs?

Reply:
ah no breaking API changes, nvm

https://github.com/pyca/bcrypt/#400

Reply:
The current fun problem is "No space left on device". Going to try splitting the build into multiple jobs (one per image)

Reply:
splitting will be difficult, i think you'd need regctl to merge and push a multi-arch image

Reply:
do you have a link to the failing build/branch?

Reply:
https://github.com/stevearc/pypicloud-docker/runs/8019057326?check_suite_focus=true#step:6:3257

Reply:
an alternative: https://github.com/docker/setup-buildx-action/blob/master/README.md#max-parallelism

although no space left can probably be avoided 'properly' as this should not be a heavy build

Reply:
Looks like most of the builds finish pretty quickly; it's just alpine linux on arm64 that's heavy. Kind of expected, since you can't use most binary packages and instead have to compile almost everything yourself https://github.com/stevearc/pypicloud-docker/runs/8039880052

Reply:
ah yes, alpine... always a pain especially for python projects 😅 we once tried it at work cuz of 'smaller images' fetish, ended up quickly reverting to debian slim

Reply:
Finally managed to build and release 1.3.9 images! I ended up disabling the arm build on alpine.

Reply:
whoop whoop!! 🎉 thanks so much for your perseverance 🙏